bin/pycbc_live

```diff
@@ -918,10 +927,17 @@ with ctx:
 
         for ifo in ifos:
             results[ifo] = False
-            status = data_reader[ifo].advance(valid_pad, timeout=args.frame_read_timeout)
-
-            if status is True:
+            status = data_reader[ifo].advance(
+                valid_pad,
+                timeout=args.frame_read_timeout
+            )
+            if status and psd_count[ifo] == 0:
                 status = data_reader[ifo].recalculate_psd()
+                psd_count[ifo] = args.psd_recompute_length - 1
```

Comment:
This seems ok to me, I am just a bit worried by a user setting `--psd-recompute-length` to an invalid value like -1. What happens then, does the code crash or does it *appear* to work and produce nonsensical results? Can I suggest adding and using a function like `pycbc.types.optparse.positive_float`, for integers?